packages/amplify-e2e-core/src/nexpect-reporter.js

```diff
@@ -76,19 +88,24 @@ class AmplifyCLIExecutionReporter {
           const recordingWithPath = recordings.map((r, index) => {
             // the first command is always 'amplify', but r.cmd is the full path to the cli.. so this is more readable
             const commandAndParams = ['amplify'];
-            if(r.params){
-              commandAndParams.push(...r.params);
+            if (r.params) {
+              commandAndParams.push(...r.params.map(p => {
+                if (p.length > 2 && p.startsWith('C:')) {
+                  return ''; // windows - skip this param because its the full amplify exe path
+                }
+                return p;
+              }));
             }
             let sanitizedSections = [];
-            for(let section of commandAndParams){
+            for (let section of commandAndParams) {
               // this ensures only alphanumeric values are in the file name
               sanitizedSections.push(section.replace(/[^a-z0-9]/gi, '_').toLowerCase());
             }
             let suffix = sanitizedSections.join('_');
-            if(suffix.length > 30){
-              suffix = suffix.substring(0, 30);
+            if (suffix.length > 20) {
+              suffix = suffix.substring(0, 20);
             }
-            const castFile = `${new Date().getTime()}_${index}_${suffix}.cast`;
+            const castFile = `${testName}_${index}_${suffix}.cast`;
```

Comment:
This change works much better with concurrent tests - and looks awesome below:
![image](https://user-images.githubusercontent.com/110861985/210117089-8bbf1e2e-fc0e-4116-9a65-afd61a03307d.png)